src/game_interpreter.cpp

```diff
@@ -4626,6 +4663,340 @@ bool Game_Interpreter::CommandManiacSetGameOption(lcf::rpg::EventCommand const&)
+			case 18: //.member[a].desc
+				if (args[2]) {
+					result = static_cast<Game_Strings::Str_t>(Main_Data::game_party->GetActor(args[1])->GetTitle());
+				}
+				else {
+					args[1] = Main_Data::game_party->GetActor(args[1])->GetId();
+					result = static_cast<Game_Strings::Str_t>(lcf::ReaderUtil::GetElement(lcf::Data::actors, args[1])->title);
+				}
+				break;
+			}
+			break;
+		case 6: //Concatenate (cat) <fn(int id_or_length_a, int id_or_length_b, int id_or_length_c)>
+		{
+			int pos = 0;
+			std::string op_string;
+			for (int i = 0; i < 3; i++) {
+				op_string += static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[i], modes[i], &pos));
+			}
+			result = (Game_Strings::Str_t)op_string;
+			break;
+		}
+		case 7: //Insert (ins) <fn(string base, int index, string insert)>
+		{
+			int pos = 0;
+			std::string base, insert;
+
+			args[1] = Main_Data::game_variables->GetWithMode(args[1], modes[1]);
+			base =    static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[0], modes[0], &pos));
+			insert =  static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[2], modes[2], &pos));
+
+			result = (Game_Strings::Str_t)base.insert(args[1], insert);
+			break;
+		}
+		case 8: //Replace (rep) <fn(string base, string search, string replacement)>
+		{
+			int pos = 0;
+			std::string base, search, replacement;
+
+			base =        static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[0], modes[0], &pos));
+			search =      static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[1], modes[1], &pos));
+			replacement = static_cast<std::string>(Main_Data::game_strings->GetWithModeAndPos(com.string, args[2], modes[2], &pos));
+
+			std::size_t index = base.find(search);
+			while (index != std::string::npos) {
+				base.replace(index, search.length(), replacement);
+				index = base.find(search, index + replacement.length());
+			}
+
+			result = static_cast<Game_Strings::Str_t>(base);
+			break;
+		}
+		case 9: //Substring (subs) <fn(string base, int index, int size)>
+			args[1] = Main_Data::game_variables->GetWithMode(args[1], modes[1]);
+			args[2] = Main_Data::game_variables->GetWithMode(args[2], modes[2]);
+			result = static_cast<Game_Strings::Str_t>(static_cast<std::string>(Main_Data::game_strings->GetWithMode(com.string, args[0], modes[0])).substr(args[1], args[2]));
+			break;
+		case 10: //Join Variables (join) <fn(string delimiter, int var_id, int size)>
+		{
+			std::string op_string = "";
+			std::string delimiter = (std::string)Main_Data::game_strings->GetWithMode(com.string, args[0], modes[0]);
+			// mode of arg[1] is weird, seems offset by +2, not sure what the intent is
+			args[1] = Main_Data::game_variables->GetWithMode(args[1], modes[1] - 2);
+			args[2] = Main_Data::game_variables->GetWithMode(args[2], modes[2]);
```

Comment:
There is already ``ValueOrVariable(mode, val)`` available here in ``Game_Interpreter``. No need to create a new method in ``game_variables``.

For string its imo fine as this is new and there are far less usages of this across the interpreter... (yet)

Reply:
Yep, I didn't realize this existed already until I went and sniffed around the StringPics code... I can refactor this away no problem.